### ISSUE TITLE
Error thrown when bound fields have been removed without blur occuring

### ISSUE THREAD

**Issue by jsobell** [aurelia/validation#293]
[https://gist.run/?id=7a4d4e8cfb84e2190cdd8d29fd133d5a](https://gist.run/?id=7a4d4e8cfb84e2190cdd8d29fd133d5a)
If you enter an email and password, then press 'Enter' to trigger the submit, the controller tries to validate a binding that no longer exists and throws an error.
Need to check to see that the binding still exists before calling.
Nice catch @starr0stealer

**Comment by starr0stealer** [comment]
https://gist.run/?id=87c111ed999bbe57b5f35bd8886c51e5

In this updated Gist-Run I added an ID to the `<button>`, and added `document.getElementById("submitBtn").focus();` into the ViewModel. Doing this the "error" no longer happens.

```
export class Login { 
  submit() {
    document.getElementById("submitBtn").focus(); // error doesn't happen if you first unFocus
    this.controller.validate();
    this.aurelia.setRoot('app'); // this results in another validation ran, I think because Blur
  }
}
```

What I think is happening, is when the User fills in the form and presses "Enter" on their keyboard, this triggers the submission of the form. Then `setRoot()` begins loading in the new View, which results in a `blur` which triggers the validation.

**Comment by starr0stealer** [comment]
I found that `setRoot()` did trigger `Blur` defined below.
https://github.com/aurelia/validation/blob/master/src/validate-binding-behavior.ts#L62
As well as `validate-binding-behavior.ts -> unbind`, but `binding.validateBlurHandler` is executed first. By the time `binding` hits `validation-controller.ts -> validateBinding` `binding.source` is `null`, which results in a `Race Condition`.

In #294 I added a `null` check within `validation-controller.ts -> validateBinding`. I took this route because of the race condition and not knowing much about `aurelia/task-queue`. It may be possible to kill the queue during `unbind` and prevent the error that way, or even a catch.
